src/textual/message_pump.py

```diff
@@ -29,6 +32,32 @@ class MessagePumpClosed(Exception):
     pass
 
 
+T = TypeVar("T")
+
+
+def _is_coroutine_callable(call: Callable) -> bool:
+    if inspect.isroutine(call):
+        return inspect.iscoroutinefunction(call)
+    if inspect.isclass(call):
+        return False
+    call = getattr(call, "__call__", None)
+    return inspect.iscoroutinefunction(call)
```

Comment:
I haven't added tests for the cases of a callable class and such. But this is [ripped from FastAPI](https://github.com/tiangolo/fastapi/blob/1760da0efa55585c19835d81afa8ca386036c325/fastapi/dependencies/utils.py#L426-L432) so I'm fairly certain we can just go with it for now

Reply:
Added 😄